### PR TITLE
Ascent Ci for ReSolve and Github

### PR DESCRIPTION
**Pull request by ryandanehy**
Creates Github actions to create a mirror on ORNL Gitlab so that we can run test on ascents and the results posted back to Github. Currently the CI is setup so that only the branch pushed is mirrored across to it corresponding ORNL's gitlab branch. Therefore Gitlab and Github have the potential to not be the same, and if a branch exists on github but not gitlab that will cause an error. However, once everything has been moved from gitlab to github. We will then mirror the entire repo so that gitlab ReSolve = Github ReSolve.

### CODE REVIEW

**Comment by ryandanehy** [comment]
Okay a bug has just now appeared in the PR vs a normal push. So needs more work.

File: .github/pnnl_mirror.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: PNNL Mirror
+
+# triggers a github action everytime there is a push or mr
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+      - main
```
Comment (pelesh) [inline .github/pnnl_mirror.yaml:9]: I would trigger it on push to any branch.

**Comment by ryandanehy** [comment]
I believe this is ready to go

File: .github/workflows/ornl_ascent_mirror.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: ORNL Ascent Mirror
+
+# triggers a github action everytime there is a push or mr
+on:
+  pull_request:
+  push:
+   branches:
+     - develop
+     - main
+
```
Comment (pelesh) [inline .github/workflows/ornl_ascent_mirror.yaml:10]: I would say trigger at each push to any branch.

File: .github/workflows/ornl_ascent_mirror.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            PUSH_ALL_REFS: "false"
+            GIT_REF: ${{ steps.extract_branch.outputs.branch }}
+            GIT_PUSH_ARGS: '--tags --force --push-option=ci.skip'
+            REMOTE: 'https://code.ornl.gov/ecpcitest/exasgd/resolve'
+            GIT_USERNAME:  ${{ secrets.GIT_USER }}
+            GIT_PASSWORD:  ${{ secrets.ORNL_ASCENT_GIT_PASSWORD }}
+
+      #  Trigger CI pipeline since it was skipped in the above push
+       - name: Trigger Pipeline
+         run: curl -X POST -F token=${{ secrets.ORNL_ASCENT_PIPELINE_TRIGGER }} -F ref=${{ steps.extract_branch.outputs.branch }} https://code.ornl.gov/api/v4/projects/13292/trigger/pipeline
```
Comment (pelesh) [inline .github/workflows/ornl_ascent_mirror.yaml:42]: Add new line at the end of the file:
```suggestion
       - name: Trigger Pipeline
         run: curl -X POST -F token=${{ secrets.ORNL_ASCENT_PIPELINE_TRIGGER }} -F ref=${{ steps.extract_branch.outputs.branch }} https://code.ornl.gov/api/v4/projects/13292/trigger/pipeline

```

**Review by pelesh — COMMENTED** [review]
I am okay with this for al long as CI pipeline istriggered on push to any branch.

File: .github/pnnl_mirror.yaml (hunk 1 above)
Comment (ryandanehy) [reply .github/pnnl_mirror.yaml:9]: The problem with that is it ends up creating two pipeline triggers for PR. Cause it technically a push to develop and PR.  We do the same for Exago.

Comment (ryandanehy) [reply .github/pnnl_mirror.yaml:9]: I originally tried to do push, but as you can see here there way to duplicate ci runs. https://code.ornl.gov/ecpcitest/exasgd/resolve/-/pipelines/468989
https://code.ornl.gov/ecpcitest/exasgd/resolve/-/pipelines/468988

**Review by pelesh — APPROVED** [review]

File: .github/workflows/ornl_ascent_mirror.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: ORNL Ascent Mirror
+
+# triggers a github action everytime there is a push or mr
+on:
+  push:
```
Comment (cameronrutherford) [inline .github/workflows/ornl_ascent_mirror.yaml:5]: On push seems fine, but there will be duplicate pipelines for PRs and pushes

File: .github/pnnl_mirror.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+name: PNNL Mirror
```
Comment (cameronrutherford) [inline .github/pnnl_mirror.yaml:1]: This file does nothing at the moment

**Pull request merged**